nanoprintf.h

```diff
@@ -17,9 +17,15 @@
 #endif
 
 #if defined(__clang__) || defined(__GNUC__) || defined(__GNUG__)
+  #define NPF_NOINLINE __attribute__((noinline))
   #define NPF_PRINTF_ATTR(FORMAT_INDEX, VARGS_INDEX) \
     __attribute__((format(printf, FORMAT_INDEX, VARGS_INDEX)))
 #else
+  #ifdef _MSC_VER
+    #define NPF_NOINLINE __declspec(noinline)
+  #else
+    #define NPF_NOINLINE
+  #endif
```

Comment:
I try to keep this top part of the header as the "public" API, since this is where users look. Since `NPF_NOINLINE` is an implementation detail and not used in the public portion of the file, could I ask you to move it down to the implementation section please?

Reply:
Good point! I will move it.